internal/testrunner/runners/system/runner.go

```diff
@@ -247,26 +262,32 @@ func (r *runner) run() ([]testrunner.TestResult, error) {
 
 		var multiErr multierror.Error
 		for _, hit := range results.Hits.Hits {
-			if hit.Source.Error != nil {
-				multiErr = append(multiErr, errors.New(hit.Source.Error.Message))
+			if message, err := hit.Source.GetValue("error.message"); err != common.ErrKeyNotFound {
+				multiErr = append(multiErr, errors.New(message.(string)))
+				continue
+			}
+
+			err := fieldsValidator.ValidateDocumentMap(hit.Source)
+			if err != nil {
+				multiErr = append(multiErr, err)
```

Comment:
Consider adding a `continue` here. I know it's not needed because this is the last check in the loop but  it future-proofs the code in case we add more code to the loop after this (and forget to add the `continue` here at the time).

Reply:
This is great argument, really appreciate it. Introduced of course.